### ISSUE TITLE
incorrect result for definite integral of rational trigonometric function

### ISSUE THREAD

**Issue by aoz** [sympy/sympy#22916]
Not sure what is causing this, but here is the incorrect result I encountered:
```
>>> import sympy
>>> from sympy import *
>>> from sympy.abc import x
>>> print(sympy.__version__)
1.8
>>> integrand = cos(2*x)/(5 - 4*cos(x))
>>> integrate(integrand, (x, 0, 2*pi))
19*pi/12
```
The correct result is pi/6. 

[Here](https://www.wolframalpha.com/input/?i=integrate%5Bcos%282*x%29%2F%285-4*cos%28x%29%29%2C%7Bx%2C0%2C+2+pi%7D%5D) is a Wolfram Alpha cross-check.

Here is a numerical version, which works fine:
```
>>> N(Integral(integrand, (x, 0, 2*pi)))
0.523598775598299
>>> N(pi/6)
0.523598775598299
```

This was on my laptop, but I reproduced it on Google Colab as well, which was running SymPy version 1.7.1.

**Comment by praneethratna** [comment]
`integrate` is returning the correct result on current master.
```
>>> from sympy import *
>>> from sympy.abc import x
>>> integrand = cos(2*x)/(5 - 4*cos(x))
>>> integrate(integrand, (x, 0, 2*pi))
pi/6
```

**Comment by aoz** [comment]
Ah ok, it has been fixed then. Thanks!

**Comment by oscarbenjamin** [comment]
This was fixed in 290889447fd0e48fa8f14563ef9d5222e0e960ac from #21840.

This issue can be closed if a test is added for this case in `test_issue_21831` in `sympy/integrals/tests/test_integrals.py`.

**Comment by oscarbenjamin** [comment]
Marking as easy to fix
